HappyApp.py

```diff
@@ -34,39 +41,61 @@ def __roll(self):
         
         if len(sadir_names) % 2 != 0:
             self.sadir_left_over = sadir_names[-1]
-            sadir_pairs += sadir_names[-1]
+            sadir_pairs.append(sadir_names[-1])
 
         all_names = sadir_pairs + keva_names
 
         shuffle(all_names)
-        self.queue = Queue(all_names)
+        self.queue = MyQueue(all_names)
 
     def __popQueue(self):
-        self.queue.dequeue()
         if self.queue.isEmpty():
             self.__roll()
-            alert(self.mail,
+            Msg.alert(self.mail,
                   data=self.queue.getQueue(),
                   msg_type=MessageType.new_period)
         else:
-            alert(self.mail,
+            self.queue.dequeue()
+            Msg.alert(self.mail,
                   data=self.queue.head(), 
                   msg_type=MessageType.reminder)
 
-    def run(self, debug):
+    def __debugTime(self, time_period = 1):
+        self.curr_hour = (self.curr_hour + 1) % 24  
+        if self.curr_hour == 0:
+            self.curr_day = (self.curr_day + 1) % 7 
+        sleep(time_period)
+        return f"Today is: {self.curr_day}, Hour: {self.curr_hour}."
+
+    def run(self):
         while True:
 
-            if isToday(day.saturday, _9PM):
-                alert(self.queue.head(), MessageType.reminder)
+            if Msg.isToday(day.wednesday, _1PM,
```

Comment:
I feel like this part has a lot of code that is repeating itself, what if you refactor this to use a dictionary instead of a lot of 'if' statements that differ only by the printing value?